simtools/data_model/validate_data.py

```diff
@@ -55,11 +55,15 @@ def validate_and_transform(self):
 
         """
 
-        self.validate_data_file()
+        if self.data_file_name:
+            self.validate_data_file()
         if isinstance(self.data, dict):
             self._validate_data_dict()
-        else:
+        elif isinstance(self.data_table, Table):
             self._validate_data_table()
+        else:
+            self._logger.error("No data or data table to validate")
+            raise TypeError
```

Comment:
I would include the raise statement in the docstring

Reply:
Fixed.